public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -239,10 +238,20 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     }
   }
 
+  waitingForDatasourceToLoad = (): boolean => {
+    // if we not yet have loaded the datasource in state the
+    // ds in props and the ds in state will have different values.
+    return this.props.dataSource.uid !== this.state.datasource?.uid;
+  };
+
   renderPluginEditor = () => {
     const { query, onChange, queries, onRunQuery, onAddQuery, app = CoreApp.PanelEditor, history } = this.props;
     const { datasource, data } = this.state;
 
+    if (this.waitingForDatasourceToLoad()) {
+      return null;
+    }
+
```

Comment:
Should we add a quick test-case for this in [QueryEditorRow.test.ts](https://github.com/grafana/grafana/blob/b3087cfcd1d1e2e08538e0c8414daaac3470286e/public/app/features/query/components/QueryEditorRow.test.ts#L0-L1)?

Reply:
I gave it a go but that class is a mess so decided to continue without adding a test. Sorry about this 😢